src/Css/StyleRule.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\Css;
+
+use Sabberworm\CSS\Property\Selector;
+use Sabberworm\CSS\RuleSet\DeclarationBlock;
+
+/**
+ * This class represents a CSS style rule, including selectors, a declaration block, and an optional containing at-rule.
+ *
+ * @internal
+ */
+class StyleRule
+{
+    /**
+     * @var DeclarationBlock
+     */
+    private $declarationBlock;
+
+    /**
+     * @var string
+     */
+    private $containingAtRule;
+
+    /**
+     * @param DeclarationBlock $declarationBlock
+     * @param string $containingAtRule e.g. `@media screen and (max-width: 480px)`
+     */
+    public function __construct(DeclarationBlock $declarationBlock, string $containingAtRule = '')
+    {
+        $this->declarationBlock = $declarationBlock;
+        $this->containingAtRule = \trim($containingAtRule);
+    }
+
+    /**
+     * @return array<int, string> the selectors, e.g. `["h1", "p"]`
+     */
+    public function getSelectors(): array
+    {
+        /** @var array<int, Selector> $selectors */
+        $selectors = $this->declarationBlock->getSelectors();
+        return \array_map(
+            static function (Selector $selector): string {
+                return (string)$selector;
+            },
+            $selectors
+        );
+    }
+
+    /**
+     * @return string the CSS declarations, separated and followed by a semicolon, e.g., `color: red; height: 4px;`
+     */
+    public function getDeclarationAsText(): string
+    {
+        return \implode(' ', $this->declarationBlock->getRules());
+    }
+
+    /**
+     * Checks whether the declaration block has at least one declaration.
+     */
+    public function hasAtLeastOneDeclaration(): bool
+    {
+        return $this->declarationBlock->getRules() !== [];
+    }
+
+    /**
+     * @returns string e.g. `@media screen and (max-width: 480px)`, or an empty string
+     */
+    public function getContainingAtRule(): string
+    {
+        return $this->containingAtRule;
+    }
+
+    /**
+     * Checks whether the containing at-rule is non-empty and has any non-whitespace characters.
```

Comment:
Indeed it won't have just whitespace because it is `trim`med in the constructor.  And I think it is implied from the DocBlock for `getContainingAtRule` that this won't return a string containing only whitespace.  I also note this is well tested 👍

Reply:
I'm note quite sure whether you're praising the method name or suggesting a change … :confused:

Reply:
This was more a comment on the internal method logic with `!== ''`, which could result in returning `true` for an at-rule consisting of only whitespace, if it were not `trim`med at some point (which it is).  But if it were to, the test `hasContainingAtRuleForEmptyContainingAtRuleReturnsFalse` would fail anyway, as would `getContainingAtRuleTrimsContainingAtRule`.

Reply:
Ah, I understand.

I try to code and name the tests in a way that the assume as little as possible about the way the subject handles its data internally (i.e., to write the tests as black-boxy as possible).